### ISSUE TITLE
[CRD-v1] Try to enforce the Connect topic names or make better defaults

### ISSUE THREAD

**Issue by scholzj** [strimzi/strimzi-kafka-operator#10075]
When running multiple Connect or MM2 clusters, the common issue is that the users have misconfigured the names of the topics used by the Connect cluster, because the defaults have a static value independent on the namespace / cluster name.

We should consider if we can improve this as part of the migration to v1 CRDs. The CRD migration might give us an opportunity to get all users to configure the topics in .spec.config and make the config required? Another option might be to have a better default names for the topics, but that would likely need to include namespace, deployment type (MM2/Connect) and cluster name.Anr even that might not work well across Kube clusters

**Comment by scholzj** [comment]
Discussed on the community call on 16.5.2024: This would be useful to have. But we should have a proposal to clarify how will we actually address this.

**Comment by maakuth** [comment]
Here's a good reference for all the internal topics that MM2 uses: https://developers.redhat.com/articles/2023/11/13/demystifying-kafka-mirrormaker-2-use-cases-and-architecture#mirrormaker_2_internal_topics

I think each of them should be in strimzi's tighter leash.

**Comment by scholzj** [comment]
Note (to my self?):
* This can be probably done for `KafkaConnect`:
    * Make the `.spec.config` section required
    * Use CEL validation (+ lkely additional in-code validation to avoid CEL bypass) to enforce the topic and group names
* But can it be done for MM2?
    * In MM2, this configuration is part of one of the _cluster_ configurations
    * It is non-trivial to find out which one should it be as it is driven by `.spec.connectName`
    * => Se we might not be able to enforce this easily on the CRD level

To be decided:
* If we cannot do this reasonably well for MM2, do we want to do it for Connect only? Or do we skip it completely?
* Should we - for MM2 - use the v1 conversion to add it to the MM2 CRs and then enforce it in the operator only? (i.e., the MM2 CR would be accepted without it, but the operator will throw `InvalidResourceException`?)

**Comment by scholzj** [comment]
Discussed on a call on 8.9.2025: We want to deal with the Connect even if we cannot solve it for MM2. @katheris will open a separate issue to discuss possible improvements for the MM2 resource itself.

**Comment by scholzj** [comment]
Another plot twist 😞. It seems CEL validation cannot be used to unstructures objects such as our `config` fields. So I think the only option we really have is to have a dedicated fields for the important options. E.g.:

```yaml
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaConnect
metadata:
  name: my-connect-cluster
spec:
  replicas: 1
  bootstrapServers: my-cluster-kafka-bootstrap:9093
  groupId: connectCluster
  offsetStorageTopicName: connect-cluster-offsets
  configStorageTopicName: connect-cluster-configs
  statusStorageTopicName: connect-cluster-status
  config:
    # -1 means it will use the default replication factor configured in the broker
    config.storage.replication.factor: -1
    offset.storage.replication.factor: -1
    status.storage.replication.factor: -1
```

These new fields will be required in `v1`. Because the operator cannot really distinguish `v1` and `v1beta2` resources, it will be initially enforced only through the CRD schema validation and only once `v1beta2` support is dropped, we will enforce it int he operator as well. Proposal should follow to confirm this. Hopefully, I don't find yet another bunch of issues when I prototype it 🤦 .

**Comment by scholzj** [comment]
Covered in proposal https://github.com/strimzi/proposals/pull/176
